dojo/tools/jfrogxray/parser.py

```diff
@@ -0,0 +1,113 @@
+import json
+from dojo.models import Finding
+
+
+class XrayJSONParser(object):
+    def __init__(self, json_output, test):
+
+        tree = self.parse_json(json_output)
+
+        if tree:
+            self.items = [data for data in self.get_items(tree, test)]
+        else:
+            self.items = []
+
+    def parse_json(self, json_output):
+        try:
+            tree = json.load(json_output)
+        except:
+            raise Exception("Invalid format")
+
+        return tree
+
+    def get_items(self, tree, test):
+        items = {}
+        if 'data' in tree:
+            vulnerabilityTree = tree['data']
+
+            for node in vulnerabilityTree:
+
+                item = get_item(node, test)
+
+                title_cve = "No CVE"
+                if 'cves' in node.get('component_versions').get('more_details'):
+                    if 'cve' in node.get('component_versions').get('more_details').get('cves')[0]:
+                        title_cve = node.get('component_versions').get('more_details').get('cves')[0].get('cve')
+
+                unique_key = node.get('id') + node.get('summary') + node.get('provider') + node.get('source_comp_id') + \
+                    title_cve
+                items[unique_key] = item
+
+        return list(items.values())
+
+
+def get_item(vulnerability, test):
+    print(vulnerability)
```

Comment:
Maybe use a logger here?

Reply:
My bad, it should not even be here. This is when I was quickly debugging. I will remove it.